client/asset/dcr/dcr.go

```diff
@@ -4524,6 +4569,123 @@ func (dcr *ExchangeWallet) EstimateSendTxFee(address string, sendAmount, feeRate
 	return finalFee, isValidAddress, nil
 }
 
+func (dcr *ExchangeWallet) isInternal() bool {
+	return dcr.walletType == walletTypeSPV
+}
+
+func (dcr *ExchangeWallet) StakeStatus() (*asset.TicketStakingStatus, error) {
+	if !dcr.connected.Load() {
+		return nil, errors.New("not connected, login first")
+	}
+	sdiff, err := dcr.wallet.StakeDiff(dcr.ctx)
+	if err != nil {
+		return nil, err
+	}
+	isRPC := !dcr.isInternal()
+	var vspURL string
+	if !isRPC {
+		if v := dcr.vspV.Load(); v != nil {
+			vspURL = v.(*vsp).Url
+		}
+	}
+	tickets, err := dcr.wallet.Tickets(dcr.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving tickets: %w", err)
+	}
+	voteChoices, tSpendPolicy, treasuryPolicy, err := dcr.wallet.VotingPreferences(dcr.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving stances: %w", err)
+	}
+	return &asset.TicketStakingStatus{
+		TicketPrice: uint64(sdiff),
+		VSP:         vspURL,
+		IsRPC:       isRPC,
+		Tickets:     tickets,
+		Stances: asset.Stances{
+			VoteChoices:    voteChoices,
+			TSpendPolicy:   tSpendPolicy,
+			TreasuryPolicy: treasuryPolicy,
+		},
+	}, nil
+}
+
+func vspInfo(url string) (*vspdjson.VspInfoResponse, error) {
+	suffix := "/api/v3/vspinfo"
+	path, err := neturl.JoinPath(url, suffix)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := http.Get(path)
+	if err != nil {
+		return nil, fmt.Errorf("http get error: %v", err)
+	}
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var info vspdjson.VspInfoResponse
+	err = json.Unmarshal(b, &info)
+	if err != nil {
+		return nil, err
+	}
+	return &info, nil
+}
+
+// SetVSP sets the VSP provider. Ability to set should be checked with CanSetVSP
+// first. Part of the asset.TicketBuyer interface.
+func (dcr *ExchangeWallet) SetVSP(url string) error {
+	if !dcr.isInternal() {
+		return errors.New("cannot set vsp for external wallet")
+	}
+	info, err := dcr.vspInfo(url)
+	if err != nil {
+		return err
+	}
+	v := vsp{
+		Url:           url,
+		PubKey:        base64.StdEncoding.EncodeToString(info.PubKey),
+		FeePercentage: info.FeePercentage,
+	}
+	b, err := json.Marshal(&v)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(dcr.dir, vspFileName), b, 0666); err != nil {
+		return err
+	}
+	dcr.vspV.Store(&v)
+	return nil
+}
+
+// PurchaseTickets purchases n amout of tickets. Ability to purchase should be
+// checked with CanPurchaseTickets. Part of the asset.TicketBuyer interface.
```

Comment:
Yeah. The original plan was to check the wallet config before purchasing over rpc, and we would deny if not using a vsp. But, giving up on that since it seems we cannot get the config. User will just have to know what they are doing. Anyway, removing the reference to the removed method.